testing/trino-product-tests/src/main/java/io/trino/tests/jdbc/TestExternalAuthorizerOAuth2.java

```diff
@@ -0,0 +1,126 @@
+import io.trino.tempto.ProductTest;
+import io.trino.tests.TpchTableResults;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.remote.RemoteWebDriver;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Properties;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static io.trino.tempto.assertions.QueryAssert.assertThat;
+import static io.trino.tempto.query.QueryResult.forResultSet;
+import static io.trino.tests.TestGroups.OAUTH2;
+import static org.openqa.selenium.support.ui.ExpectedConditions.elementToBeClickable;
+
+public class TestExternalAuthorizerOAuth2
+        extends ProductTest
+{
+    @Inject
+    @Named("databases.presto.jdbc_url")
+    String jdbcUrl;
+
+    private final ExecutorService executor = Executors.newSingleThreadExecutor();
+
+    @Test(groups = OAUTH2)
+    public void shouldAuthenticateAndExecuteQuery()
+            throws SQLException
+    {
+        prepareSeleniumHandler();
+        Properties properties = new Properties();
+        properties.setProperty("user", "test");
+        try (Connection connection = DriverManager.getConnection(jdbcUrl, properties);
+                PreparedStatement statement = connection.prepareStatement("select * from tpch.tiny.nation");
+                ResultSet results = statement.executeQuery()) {
+            assertThat(forResultSet(results)).matches(TpchTableResults.PRESTO_NATION_RESULT);
+        }
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void clean()
+    {
+        executor.shutdown();
+    }
+
+    private void prepareSeleniumHandler()
+    {
+        TrinoDriverUri.setRedirectHandler(uri -> executor.execute(() -> {
+            WebDriver driver = getWebDriver();
+            driver.get(uri.toString());
+            WebDriverWait wait = new WebDriverWait(driver, 5);
+            submitCredentials(driver, "foo@bar.com", "foobar", wait);
+            giveConsent(driver, wait);
+        }));
+    }
+
+    private WebDriver getWebDriver()
+    {
+        ChromeOptions options = new ChromeOptions();
+        options.setAcceptInsecureCerts(true);
+        return new RemoteWebDriver(getWebDriverUrl(), options);
+    }
+
+    private URL getWebDriverUrl()
+    {
+        try {
+            return new URL("http", "selenium-chrome", 7777, "/wd/hub");
+        }
+        catch (MalformedURLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void submitCredentials(WebDriver driver, String email, String password, WebDriverWait wait)
```

Comment:
Please add verbose logging here and to the method below. Even on INFO that we are passing password, we are waiting etc.

Reply:
Have you tried verbose logging? https://chromedriver.chromium.org/logging